src/gf.c

```diff
@@ -609,6 +615,19 @@ static jl_lambda_info_t *cache_method(jl_methtable_t *mt, union jl_typemap_t *ca
         // in the method cache to prevent anything else from matching this entry
         origtype = NULL;
     }
+    if (makesimplesig && origtype == NULL) {
+        // reduce the complexity of rejecting this entry in the cache
```

Comment:
See the example in the PR. This lets us make an expensive cache entry like (<: Function), while still using the fast-path to reject it (although the (<: Any) it builds isn't a particularly useful example since it only has one argument)